mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala

```diff
@@ -0,0 +1,740 @@
+  final val docConcentration = new DoubleArrayParam(this, "docConcentration",
+    "Concentration parameter (commonly named \"alpha\") for the prior placed on documents'" +
+      " distributions over topics (\"theta\").", validDocConcentration)
+
+  /** Check that the docConcentration is valid, independently of other Params */
+  private def validDocConcentration(alpha: Array[Double]): Boolean = {
+    if (alpha.length == 1) {
+      alpha(0) == -1 || alpha(0) >= 1.0
+    } else if (alpha.length > 1) {
+      alpha.forall(_ >= 1.0)
+    } else {
+      false
+    }
+  }
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getDocConcentration: Array[Double] = $(docConcentration)
+
+  /**
+   * Alias for [[getDocConcentration]]
+   * @group getParam
+   */
+  @Since("1.6.0")
+  def getAlpha: Array[Double] = getDocConcentration
+
+  /**
+   * Concentration parameter (commonly named "beta" or "eta") for the prior placed on topics'
+   * distributions over terms.
+   *
+   * This is the parameter to a symmetric Dirichlet distribution.
+   *
+   * Note: The topics' distributions over terms are called "beta" in the original LDA paper
+   * by Blei et al., but are called "phi" in many later papers such as Asuncion et al., 2009.
+   *
+   * If set to -1, then topicConcentration is set automatically.
+   *  (default = -1 = automatic)
+   *
+   * Optimizer-specific parameter settings:
+   *  - EM
+   *     - Value should be > 1.0
+   *     - default = 0.1 + 1, where 0.1 gives a small amount of smoothing and +1 follows
+   *       Asuncion et al. (2009), who recommend a +1 adjustment for EM.
+   *  - Online
+   *     - Value should be >= 0
+   *     - default = (1.0 / k), following the implementation from
+   *       [[https://github.com/Blei-Lab/onlineldavb]].
+   * @group param
+   */
+  @Since("1.6.0")
+  final val topicConcentration = new DoubleParam(this, "topicConcentration",
+    "Concentration parameter (commonly named \"beta\" or \"eta\") for the prior placed on topic'" +
+      " distributions over terms.", (beta: Double) => beta == -1 || beta >= 0.0)
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getTopicConcentration: Double = $(topicConcentration)
+
+  /**
+   * Alias for [[getTopicConcentration]]
+   * @group getParam
+   */
+  @Since("1.6.0")
+  def getBeta: Double = getTopicConcentration
+
+  /**
+   * Optimizer or inference algorithm used to estimate the LDA model, specified as a
+   * [[LDAOptimizer]] type.
+   * Currently supported:
+   *  - Online Variational Bayes: [[OnlineLDAOptimizer]] (default)
+   *  - Expectation-Maximization (EM): [[EMLDAOptimizer]]
+   * @group param
+   */
+  @Since("1.6.0")
+  final val optimizer = new Param[LDAOptimizer](this, "optimizer", "Optimizer or inference" +
+    " algorithm used to estimate the LDA model")
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getOptimizer: LDAOptimizer = $(optimizer)
+
+  // Developers should override these setOptimizer() methods.  These are defined here to
```

Comment:
The main point was to provide the setOptimizer taking a String here so that subclasses don't repeat it.  But that's actually not needed since LDAModel does not need this.  I'll move them to LDA.